exes/mqtt-bridge/src/client.cpp

```diff
@@ -190,18 +190,32 @@ auto client<client_t, config_t>::wait_for_payloads(
 
     logger_.trace("Publish packet received, parsing...");
 
-    auto* publish_packet = publish_recv.template get_if<async_mqtt::v5::publish_packet>();
-
-    if (publish_packet == nullptr) {
-      logger_.error("Received packet is not a PUBLISH packet");
+    if (auto* publish_packet = publish_recv.template get_if<async_mqtt::v5::publish_packet>()) {
+      for (uint64_t i = 0; i < publish_packet->payload().size(); i++) {
+        logger_.trace("Received PUBLISH packet. Parsing payload...");
+        process_payload(publish_packet->payload()[i], *publish_packet);
+      }
+    } else {
+      if (auto* puback_packet = publish_recv.template get_if<async_mqtt::v5::puback_packet>()) {
+        logger_.error("Received packet is a puback packet");
```

Comment:
Covered above?

Reply:
This if statement is covering a puback packet, not a publish packet.